applications/crossbar/src/modules/provisioner_v5.erl

```diff
@@ -349,14 +349,37 @@ settings_keys(Assoc, KeyKind, JObj) ->
         LineKey -> kz_json:set_value(<<"account">>, LineKey, Keys)
     end.
 
+-spec get_label(binary(), binary()) -> binary().
+get_label(AccountId, DocId) ->
+    AccountDb = kz_util:format_account_id(AccountId, 'encoded'),
+    Doc = kz_datamgr:open_cache_doc(AccountDb, DocId),
+
+    get_label(Doc).
+
+-spec get_label(kz_json:object()) -> binary().
+get_label(Doc) ->
+    case {kz_json:get_value(<<"first_name">>, Doc), kz_json:get_value(<<"last_name">>, Doc)} of
+        {'undefined', 'undefined'} ->
+            kz_json:get_value(<<"name">>, Doc);
+
+        {First, 'undefined'} ->
+            First;
+
+        {'undefined', _Last} ->
+            kz_json:get_value(<<"name">>, Doc);
```

Comment:
@mark2600 this should never happen ;) but if there is no first_name then there will be no 'name'.  Should this not return Last?

Also why the space between clauses?